flink-rpc/flink-rpc-core/src/main/java/org/apache/flink/runtime/rpc/FencedRpcEndpoint.java

```diff
@@ -40,23 +35,17 @@
  */
 public abstract class FencedRpcEndpoint<F extends Serializable> extends RpcEndpoint {
 
-    private final UnfencedMainThreadExecutor unfencedMainThreadExecutor;
-    private volatile F fencingToken;
+    private final F fencingToken;
     private volatile MainThreadExecutor fencedMainThreadExecutor;
```

Comment:
It seems we no longer need this `fencedMainThreadExecutor`, as now it is no different from `RpcEndpoint#mainThreadExecutor`.

Reply:
Nice suggestion, will remove it now.